Explorer/Assets/DCL/Backpack/BackpackGridController.cs

```diff
@@ -0,0 +1,157 @@
+        private World world;
+
+        public BackpackGridController(
+            BackpackGridView view,
+            BackpackCommandBus commandBus,
+            BackpackEventBus eventBus,
+            IWeb3IdentityCache web3IdentityCache,
+            NftTypeIconSO rarityBackgrounds,
+            NFTColorsSO rarityColors,
+            NftTypeIconSO categoryIcons)
+        {
+            this.view = view;
+            this.commandBus = commandBus;
+            this.eventBus = eventBus;
+            this.web3IdentityCache = web3IdentityCache;
+            this.rarityBackgrounds = rarityBackgrounds;
+            this.rarityColors = rarityColors;
+            this.categoryIcons = categoryIcons;
+
+            usedPoolItems = new ();
+            eventBus.EquipEvent += OnEquip;
+            eventBus.UnEquipEvent += OnUnequip;
+        }
+
+        public void InjectToWorld(ref ArchSystemsWorldBuilder<World> builder, in Entity playerEntity)
+        {
+            world = builder.World;
+        }
+
+        public async UniTask InitialiseAssetsAsync(IAssetsProvisioner assetsProvisioner, CancellationToken ct)
+        {
+            BackpackItemView backpackItem = (await assetsProvisioner.ProvideInstanceAsync(view.BackpackItem, ct: ct)).Value;
+
+            gridItemsPool = new ObjectPool<BackpackItemView>(
+                () => CreateBackpackItem(backpackItem),
+                _ => { },
+                defaultCapacity: 16
+            );
+        }
+
+        private void SetGridElements(IWearable[] gridWearables)
+        {
+            ClearPoolElements();
+            for (var i = 0; i < gridWearables.Length; i++)
+            {
+                BackpackItemView backpackItemView = gridItemsPool.Get();
+                usedPoolItems.Add(gridWearables[i].GetUrn(), backpackItemView);
+
+                backpackItemView.OnSelectItem += SelectItem;
+                backpackItemView.ItemId = gridWearables[i].GetUrn();
+                backpackItemView.RarityBackground.sprite = rarityBackgrounds.GetTypeImage(gridWearables[i].GetRarity());
+                backpackItemView.FlapBackground.color = rarityColors.GetColor(gridWearables[i].GetRarity());
+                backpackItemView.CategoryImage.sprite = categoryIcons.GetTypeImage(gridWearables[i].GetCategory());
+
+                WaitForThumbnailAsync(gridWearables[i], backpackItemView).Forget();
+            }
+        }
+
+        private BackpackItemView CreateBackpackItem(BackpackItemView backpackItem)
+        {
+            BackpackItemView backpackItemView = Object.Instantiate(backpackItem, view.gameObject.transform);
+            backpackItem.EquipButton.onClick.AddListener(() => commandBus.SendCommand(new BackpackEquipCommand(backpackItemView.ItemId)));
+            return backpackItemView;
+        }
+
+        public void RequestPage(int pageNumber)
+        {
+            //Reuse params array and review types URLParameter once auth pr is merged
```

Comment:
I think it should be addressed: it does not seem heavy to implement. Moreover, the auth PR is already merged